### PR TITLE
doc: add troubleshooting for RDS maximum policy size exceeded errors

### PR DESCRIPTION
**Pull request by greedy52**

### CODE REVIEW

File: docs/pages/database-access/guides/rds.mdx [hunk 1]
```diff
@@ -209,6 +209,8 @@ $ tsh db logout postgres-rds
 
 (!docs/pages/includes/database-access/aws-troubleshooting.mdx!)
 
+(!docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx!)
```
Comment (r0mant) [inline docs/pages/database-access/guides/rds.mdx:212]: Why not just include this topic in aws-troubleshooting.mdx?

File: docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Due to [IAM and STS character
+limits](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html#reference_iam-quotas-entity-length),
+you may encounter one of the following errors in the Database Service logs, when large numbers of databases are
+registered:
+- `LimitExceeded: Maximum policy size of 2048 bytes exceeded for user <iam-user>`
+- `LimitExceeded: Maximum policy size of 10240 bytes exceeded for role <iam-role>`
+
+For reference, a user policy can maintain permissions for approximately 6 Redshift databases, or 20 RDS databases. And a
+role policy can maintain permissions for approximately 30 Redshift databases, or 100 RDS databases.
```
Comment (r0mant) [inline docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx:11]: Mention that this is due to AWS policy character length limits.

**Review by r0mant — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Review by smallinsky — COMMENTED** [review]
LGTM Thanks for the updated.

File: docs/pages/database-access/guides/rds.mdx (hunk 1 above)
Comment (greedy52) [reply docs/pages/database-access/guides/rds.mdx:212]: This error is not applicable to other AWS databases (e.g. AWS keyspaces, Elaticache, etc.). I think it's cleaner this way.

File: docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx [hunk 3]
```diff
@@ -0,0 +1,61 @@
+### Maximum policy size exceeded errors
+
+Due to [IAM and STS character
+limits](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html#reference_iam-quotas-entity-length),
+you may encounter one of the following errors in the Database Service logs, when large numbers of databases are
+registered:
```
Comment (alexfornuto) [inline docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx:6]: Removed comma, fixed hard breaks:
```suggestion
you may encounter one of the following errors in the Database Service logs when
large numbers of databases are registered:
```

File: docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+- `LimitExceeded: Maximum policy size of 2048 bytes exceeded for user <iam-user>`
+- `LimitExceeded: Maximum policy size of 10240 bytes exceeded for role <iam-role>`
+
+For reference, a user policy can maintain permissions for approximately 6 Redshift databases, or 20 RDS databases due to
+the IAM policy character limits. And a role policy can maintain permissions for approximately 30 Redshift databases, or
+100 RDS databases.
```
Comment (alexfornuto) [inline docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx:12]: Fixed line breaks, Edited so sentence doesn't start with "and":
```suggestion
For reference, a user policy can maintain permissions for approximately 6 Redshift
databases, or 20 RDS databases due to the IAM policy character limits. A role
policy can maintain permissions for approximately 30 Redshift databases, or 100
RDS databases.
```

File: docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+One way to work around this is to manually manage the IAM policy used for database connection and use a wildcard `"*"`
+for `"Resource"` to reduce the policy size:
```
Comment (alexfornuto) [inline docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx:15]: Line breaks and copy edits:
```suggestion
One way to work around this is to manually manage the IAM policy used for database
connections, and use a wildcard `"*"` for `"Resource"` to reduce the policy size:
```

File: docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+              "Effect": "Allow",
+              "Action": "rds-db:connect",
+              "Resource": "*"
+          }
+      ]
+  }
+  ```
+  </TabItem>
+  <TabItem label="Redshift">
+  ```
```
Comment (alexfornuto) [inline docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx:37]:
````suggestion
  ```json
````

File: docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+              "Resource": "*"
+          }
+      ]
+  }
+  ```
+  </TabItem>
+</Tabs>
+
+You can safely remove the inline policy created by the Database Service and the IAM permissions for the Database Service
+to `Get/Put/Delete` the user or role policy.
```
Comment (alexfornuto) [inline docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx:53]: Line breaks:
```suggestion
You can safely remove the inline policy created by the Database Service and the
IAM permissions for the Database Service to `Get/Put/Delete` the user or role policy.
```

File: docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+</Details>
+
+Another option to resolve this is to deploy [Database Access in a Highly Available (HA)
+configuration](../../database-access/guides/ha.mdx) where databases can be sharded to separate Database Services with
+different IAM roles.
+
+If the limit is exceeded for a user policy, it is also recommended to use an IAM role for the Database Service instead.
```
Comment (alexfornuto) [inline docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx:61]: Copy edits and line breaks:
```suggestion
Another option is to deploy [Database Access in a Highly Available (HA)
configuration](../../database-access/guides/ha.mdx) where databases can be sharded
to separate Database Services with different IAM roles.

If the limit is exceeded for a user policy, it is also recommended to use an IAM
role for the Database Service instead.
```

File: docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx (hunk 3 above)
Comment (greedy52) [reply docs/pages/includes/database-access/aws-troubleshooting-max-policy-size.mdx]: I had a 120 textwidth for some other changes and forgot to reset it to 80. Adjusted all paragraphs to 80 now. Thanks!

**Review by alexfornuto — APPROVED** [review]

**Pull request merged**